pages/docs/deployment/migrate-v2-to-v3.mdx

```diff
@@ -0,0 +1,223 @@
+
+#### Clickhouse vs. others
+
+We think there are many great OLAP databases out there and are sure that we could have chosen an alternative and would also succeed with it. However, here are some thoughts on alternatives:
+- Druid: Unlike Druid's [modular architecture](https://posthog.com/blog/clickhouse-vs-druid), ClickHouse provides a more straightforward, unified instance approach. Hence, it is easier for teams to manage Clickhouse in production as there are fewer moving parts. This reduces the operational burden especially for our self-hosters.
+- StarRocks: We think StarRocks is great but early. The vast amount of features in Clickhouse help us to remain flexible with our requirements while benefiting from the performance of an OLAP database.
+
+#### Building an adapter and support multiple databases
+
+We explored building a multi-database adapter to support Postgres for smaller self-hosted deployments.
+After talking to engineers and reviewing some of PostHog's [Clickhouse implementation](https://github.com/PostHog/posthog),
+we decided against this path due to its complexity and maintenance overhead.
+This allows us to focus our resources on building user features instead.
+
+### Why Redis
+
+We added a Redis instance to serve cache and queue use-cases within our stack.
+With its open source license, broad native support my major cloud vendors, and ubiquity in the industry, Redis was a natural choice for us.
+
+### Why S3/Blob Store
+
+Observability data for LLM application tends to contain large, semi-structured bodies of data to represent inputs and outputs.
+We chose S3/Blob Store as a scalable, secure, and cost-effective solution to store these large objects.
+It allows us to store all incoming events for further processing and acts as a native backup solution, as the full state
+can be restored based on the events stored there.
+
+### Why Worker Container
+
+When processing observability data for LLM applications, there are many CPU-heavy operations which block the main loop in our Node.js backend,
+e.g. tokenization and other parsing of event bodies.
+To achieve high availability and low latencies across client applications, we decided to move the heavy processing into an asynchronous worker container.
+It accepts events from a Redis queue and ensures that they are eventually being upserted into Clickhouse.
+
+
+## SDK compatibility and API changes
+
+While we aim to keep our SDKs and APIs fully backwards compatible, we have to introduce backwards incompatible changes with our update to V3.
+Certain APIs in SDK versions below version 2.0.0 are not compatible with our new backend architecture.
+Please upgrade and benefit from many performance improvements or features such as [prompt caching](https://langfuse.com/changelog/2024-02-05-sdk-level-prompt-caching).
```

Comment:
for links in our documentation, relative links are strongly preferred
- they work in development
- they use next Link which does not need a pageload to render the next page in prod